build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -1156,6 +1156,87 @@ export const handlers: Handler[] = [
 			}
 		},
 	},
+	{
+		// This rule enforces each package has an exports field in its package.json. It also verifies that the values in the
+		// exports["."] field match the ones in the main/module/types fields.
+		name: "npm-package-exports-field",
+		match,
+		handler: (file) => {
+			let json: PackageJson;
+
+			try {
+				json = JSON.parse(readFile(file));
+			} catch (err) {
+				return "Error parsing JSON file: " + file;
+			}
+
+			if (!shouldCheckExportsField(json)) {
+				return;
+			}
+
+			if (json.types === undefined && json.typings === undefined) {
+				throw new Error(
+					"The 'types' and 'typings' field are both undefined. At least one must be defined.",
+				);
+			}
+
+			if (json.main === undefined) {
+				throw new Error("The 'main' field is undefined. It must have a value.");
+			}
```

Comment:
These checks should probably be in the resolver only.

Reply:
These checks are already done in shouldCheckExportsField, so they can be removed.